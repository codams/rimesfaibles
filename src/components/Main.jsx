import React, { useState } from "react";
import styled from "styled-components";

import { WordInput } from "./WordInput";
import { WordOutput } from "./WordOutput";

import { WordsContext } from "../contexts/WordsContext";

const SectionStyled = styled.section`
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
  padding: 6vh;
  flex-grow: 1;
`;

export const Main = () => {
  const WordsProvider = WordsContext.Provider;
  const [input, setInput] = useState("");
  console.log("Built by damsauvagere@gmail.com");
  return (
    <SectionStyled>
      <WordsProvider value={{ input, setInput }}>
        <WordInput />
        <WordOutput />
      </WordsProvider>
    </SectionStyled>
  );
};
