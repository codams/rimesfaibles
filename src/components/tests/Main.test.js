import { Main } from "../Main";

import { screen, render } from "@testing-library/dom";
import TestRenderer from "react-test-renderer";
import "jest-styled-components";

test("test if display is flex", () => {
  const tree = TestRenderer.create(<Main />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule("display", "flex");
});

test("test if paddinf exist", () => {
  const tree = TestRenderer.create(<Main />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule("padding", "6vh");
});
