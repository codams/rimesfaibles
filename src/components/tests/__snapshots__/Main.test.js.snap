// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test if display is flex 1`] = `
.c2 {
  text-align: center;
  height: 8vh;
  border-radius: 2em;
  font-size: 4vw;
  color: #eb219b;
}

.c2:focus {
  border: 4px solid #eb219b;
  outline: none;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c3 {
  padding: 5px;
  margin-top: 4vh;
  height: 8vh;
  border-radius: 2em;
  background-color: #eb219b;
  font-family: "Mr Dafoe",cursive;
  color: #fff;
  -webkit-text-fill-color: #fff;
  text-shadow: -1px -1px 0 #fff;
  -webkit-filter: drop-shadow(1px 1px 1px #fff);
  font-weight: 700;
  font-size: 3vh;
}

.c5 {
  color: transparent;
  text-transform: uppercase;
  font-size: 8vw;
  background: linear-gradient( cornflowerblue 30%, #e4e4e4, darkmagenta 60%, cornflowerblue 100% );
  background-size: contain;
  -webkit-background-clip: text;
  background-clip: text;
  -webkit-text-stroke: 1px #000000;
  -webkit-filter: drop-shadow(2px 2px 1px black) drop-shadow(1px 1px 1px magenta);
  filter: drop-shadow(2px 2px 1px black) drop-shadow(1px 1px 1px magenta);
}

.c4 {
  margin-top: 3vh;
  border: 4px solid #eb219b;
  box-sizing: border-box;
  display: justify-content:center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 2vw;
  -webkit-filter: drop-shadow(2px 2px 1px #88006a) drop-shadow(1px 1px 1px magenta);
  filter: drop-shadow(2px 2px 1px #88006a) drop-shadow(1px 1px 1px magenta);
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 6vh;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

<section
  className="c0"
>
  <form
    className="c1"
    onSubmit={[Function]}
  >
    <input
      className="c2"
      id="wordInput"
      name="wordInput"
      onChange={[Function]}
      type="text"
    />
    <input
      className="c3"
      type="submit"
      value="Entre un mot pour connaitre sa rime !"
    />
  </form>
  <div
    className="c4"
    display="none"
  >
    <h4
      className="c5"
    >
      
    </h4>
  </div>
</section>
`;

exports[`test if paddinf exist 1`] = `
.c2 {
  text-align: center;
  height: 8vh;
  border-radius: 2em;
  font-size: 4vw;
  color: #eb219b;
}

.c2:focus {
  border: 4px solid #eb219b;
  outline: none;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c3 {
  padding: 5px;
  margin-top: 4vh;
  height: 8vh;
  border-radius: 2em;
  background-color: #eb219b;
  font-family: "Mr Dafoe",cursive;
  color: #fff;
  -webkit-text-fill-color: #fff;
  text-shadow: -1px -1px 0 #fff;
  -webkit-filter: drop-shadow(1px 1px 1px #fff);
  font-weight: 700;
  font-size: 3vh;
}

.c5 {
  color: transparent;
  text-transform: uppercase;
  font-size: 8vw;
  background: linear-gradient( cornflowerblue 30%, #e4e4e4, darkmagenta 60%, cornflowerblue 100% );
  background-size: contain;
  -webkit-background-clip: text;
  background-clip: text;
  -webkit-text-stroke: 1px #000000;
  -webkit-filter: drop-shadow(2px 2px 1px black) drop-shadow(1px 1px 1px magenta);
  filter: drop-shadow(2px 2px 1px black) drop-shadow(1px 1px 1px magenta);
}

.c4 {
  margin-top: 3vh;
  border: 4px solid #eb219b;
  box-sizing: border-box;
  display: justify-content:center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 2vw;
  -webkit-filter: drop-shadow(2px 2px 1px #88006a) drop-shadow(1px 1px 1px magenta);
  filter: drop-shadow(2px 2px 1px #88006a) drop-shadow(1px 1px 1px magenta);
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 6vh;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

<section
  className="c0"
>
  <form
    className="c1"
    onSubmit={[Function]}
  >
    <input
      className="c2"
      id="wordInput"
      name="wordInput"
      onChange={[Function]}
      type="text"
    />
    <input
      className="c3"
      type="submit"
      value="Entre un mot pour connaitre sa rime !"
    />
  </form>
  <div
    className="c4"
    display="none"
  >
    <h4
      className="c5"
    >
      
    </h4>
  </div>
</section>
`;
